### DISCUSSION TITLE
DAGBAG_IMPORT_TIMEOUT does not work

### DISCUSSION THREAD

**Discussion by nijkah** [apache/airflow#17991] [Q&A]
### Apache Airflow version

2.1.1

### Operating System

18.04.5 LTS (Bionic Beaver)

### Versions of Apache Airflow Providers

apache-airflow==2.1.1
apache-airflow-providers-amazon==2.0.0
apache-airflow-providers-celery==2.0.0
apache-airflow-providers-cncf-kubernetes==2.0.0
apache-airflow-providers-docker==2.0.0
apache-airflow-providers-elasticsearch==2.0.1
apache-airflow-providers-ftp==2.0.0
apache-airflow-providers-google==4.0.0
apache-airflow-providers-grpc==2.0.0
apache-airflow-providers-hashicorp==2.0.0
apache-airflow-providers-http==2.0.0
apache-airflow-providers-imap==2.0.0
apache-airflow-providers-microsoft-azure==3.0.0
apache-airflow-providers-mysql==2.0.0
apache-airflow-providers-postgres==2.0.0
apache-airflow-providers-redis==2.0.0
apache-airflow-providers-sendgrid==2.0.0
apache-airflow-providers-sftp==2.0.0
apache-airflow-providers-slack==4.0.0
apache-airflow-providers-sqlite==2.0.0
apache-airflow-providers-ssh==2.0.0

### Deployment

Official Apache Airflow Helm Chart

### What happened

We want to import a dag with 1,900 nodes with CustomOperators.

It does work up to 300 nodes.

But when we set a dag with 1,900 nodes.
It throws error like this.
![image](https://user-images.githubusercontent.com/25769408/131846372-5b2fdd3d-04d5-4a2a-9aa7-c7ffbe4f6217.png)

So we add variables to custom_values.yaml for chart like below.
```
env:
     - name: AIRFLOW__CORE__DAGBAG_IMPORT_TIMEOUT
        value: 360.0
config:
  core:
    dagbag_import_timeout: 360.0
```

But it still says, timeout is 30.s in error message.
So with interactive shell of worker pod, I checked this.

![image](https://user-images.githubusercontent.com/25769408/131846806-bcb1d4e2-49e9-4848-933c-5cbd5f78baf2.png)

Why configuration does not set?

### What you expected to happen

Although I set a configuration for dagbag_import_timeout, it still throws messages with 30.0s.

### How to reproduce

I think any dag with import timeout exceeding 30.0s can show that the message does not change despite of setting configuration for dagbag_import_timeout.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Did you restart your scheduler  after the change? and did you change it as "global" variable or "worker" one? 

I believe you should repeat your getfloat() from `scheduler` not from "worker" - and it is very likely your scheduler still uses the old value.

However I have side comment, but very important. You will have a LOT of problems if your DAGs are importing that long.

Import of > 30 seconds is probably an indication of bigger problem you have actually. It probably means that you do not follow best practices for Top-Level code: https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code  . I think you should rather focus on making sure that you do not do any heavy-lifting during importing your DAGs and move all the heaving processing to either before dag is scheduled (generating some meta-data) or to inside the "execute" methods in your operators or JINJA templating)
